viewer/Viewer.cpp

```diff
@@ -106,26 +107,55 @@ namespace Enki
 		yaw(yaw),
 		pitch(pitch)
 	{
-		
+		update(false);
 	}
 	
+	void ViewerWidget::CameraPose::update(bool trackballMode, QVector3D targetPosition, float zNear)
+	{
+		forward = QVector3D( cos(yaw)*cos(pitch), sin(yaw)*cos(pitch), sin(pitch) );
+		left = QVector3D::crossProduct(QVector3D(0,0,1), forward).normalized();
+		up = QVector3D::crossProduct(forward, left).normalized();
+
+		if(trackballMode)
+		{
+			pos.rx() = targetPosition.x() - radius*forward.x();
+			pos.ry() = targetPosition.y() - radius*forward.y();
+			altitude = targetPosition.z() + zNear*1.01 - radius*forward.z();
+		}
+	}
+
 	ViewerWidget::ViewerWidget(World *world, QWidget *parent) :
 		QGLWidget(parent),
 		timerPeriodMs(30),
-		world(world),
-		worldList(0),
-		mouseGrabbed(false),
 		camera(CameraPose(
 			QPointF(world->w * 0.5, qMax(0., world->r)),
 			qMax(qMax(world->w, world->h), world->r*2) * 0.85,
 			M_PI/2,
 			-(3*M_PI)/8
 		)),
-		wallsHeight(10),
 		doDumpFrames(false),
+		world(world),
+		worldList(0),
+		mouseGrabbed(false),
+		wallsHeight(10),
+		trackballView(false),
 		dumpFramesCounter(0)
 	{
 		initTexturesResources();
+		pointedObject = 0;
+		selectedObject = 0;
+
+		controlError1 = "object translation not avalaible in trackball mode";
```

Comment:
I can remove this feature but I think that's realy helpful when the user try to perform an action not allowed.
For camera control it's disturbing when the viewer doesn't respond to control.
And for the help string it's related to camera and picking control, so related to viewer for me.